.circleci/config.yml

```diff
@@ -87,49 +108,77 @@ jobs:
-          dir: ~/expo/packages/expo
+      - yarn:
+          command: jest --maxWorkers 1
+          working_directory: ~/expo/packages/expo
 
   babel_preset:
     executor: js
     steps:
-      - checkout
-      - set_up_environment
+      - setup
+      - update_submodules
       - restore_yarn_cache
-      - install_workspace_dependencies
+      - yarn:
+          working_directory: ~/expo
       - save_yarn_cache
-      - lint:
-          dir: ~/expo/packages/babel-preset-expo
-      - jest:
-          dir: ~/expo/packages/babel-preset-expo
+      - yarn:
+          command: lint --max-warnings 0
+          working_directory: ~/expo/packages/babel-preset-expo
+      - yarn:
+          command: jest --maxWorkers 1
+          working_directory: ~/expo/packages/babel-preset-expo
 
   home:
     executor: js
     steps:
-      - checkout
-      - set_up_environment
+      - setup
+      - update_submodules
       - restore_yarn_cache
-      - install_workspace_dependencies
+      - yarn:
+          working_directory: ~/expo
       - save_yarn_cache
-      - jest:
-          dir: ~/expo/home
+      - yarn:
+          command: jest --maxWorkers 1
+          working_directory: ~/expo/home
 
   client_ios:
     executor: mac
     steps:
-      - checkout
-      - set_up_environment
+      - install_nix
+      - setup
       - fetch_cocoapods_specs
-      - run:
-          working_directory: ~/project/tools-public
-          command: yarn install
-      - run:
+      - yarn:
           working_directory: ~/project/tools-public
-          command: PATH=$(yarn bin):$PATH ./generate-files-ios.sh
+      - run: nix run nixpkgs.nodejs-8_x -c ~/project/tools-public/generate-files-ios.sh
       - run: fastlane ios create_simulator_build
+
+  client_android:
+    executor: android
+    steps:
+      - install_nix
+      - setup
+      - yarn:
+          working_directory: ~/expo/tools-public
+      - run: nix-env -f . -iA nodejs-8_x # For some reason `nix run` doesn't work
+      - run: ~/expo/tools-public/generate-dynamic-macros-android.sh
+      - restore_gradle_cache
+      - run:
+          working_directory: ~/expo/android
+          command: ./gradlew :app:preBuild
+      - save_gradle_cache
+      - run:
+          working_directory: ~/expo/android
+          command: ./gradlew :app:assembleProdMinSdkProdKernelRelease
+      - run: echo 'export PATH="/opt/android/sdk/build-tools/27.0.3:$PATH"' >> $BASH_ENV # should match buildToolsVersion in android/app/build.gradle
+      - run: zipalign -v -p 4 android/app/build/outputs/apk/prodMinSdkProdKernel/release/app-prodMinSdk-prodKernel-release-unsigned.apk app-prod-release-unsigned-aligned.apk
+      - run: echo $ANDROID_KEYSTORE_B64 | base64 -d > exponentkeystore.jks
+      - run: apksigner sign --ks exponentkeystore.jks --ks-key-alias $ANDROID_KEY_ALIAS --ks-pass env:ANDROID_KEYSTORE_PASSWORD --key-pass env:ANDROID_KEY_PASSWORD --out exponent-release.apk app-prod-release-unsigned-aligned.apk
```

Comment:
CircleCI's UI uses the whole command as the name of the step by default (https://circleci.com/gh/expo/expo/1445) -- it might be nicer to give these steps explicit descriptions like "Sign the APK". Will leave it up to you.